src/dataframe/dataframe.jl

```diff
@@ -216,6 +214,23 @@ struct DataFrame <: AbstractDataFrame
     end
 end
 
+function _preprocess_column(col::Any, len::Integer, copycols::Bool)
+    # check for vectors first as they are most common
```

Comment:
Thanks to specialization, the order of checks shouldn't have any effect on performance.
```suggestion
```

Reply:
ah yes - after specialization it will be optimized out